### PR TITLE
signaturepdf: add service

### PR DESCRIPTION
**Pull request by DamienCassou**
### Description

New signaturepdf service. https://github.com/24eme/signaturepdf/

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

### CODE REVIEW

File: modules/services/signaturepdf.nix [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+let
+  cfg = config.services.signaturepdf;
+  extraConfigToArgs = extraConfig:
+    lib.flatten
+    (lib.mapAttrsToList (name: value: [ "-d" "${name}=${value}" ]) extraConfig);
+in {
+  meta.maintainers = [ lib.maintainers.DamienCassou ];
+
+  options.services.signaturepdf = with lib; {
+    enable = mkEnableOption "signaturepdf, the clipboard management daemon";
```
Comment (rycee) [inline modules/services/signaturepdf.nix:12]: Maybe needs a new description? I think this is not actually a clipboard management daemon?

File: modules/services/signaturepdf.nix [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 8080;
+      example = 8081;
+      description = "The port on which the application runs";
+    };
+
+    extraConfig = mkOption {
```
Comment (rycee) [inline modules/services/signaturepdf.nix:28]: I guess this should be 
```suggestion
    settings = mkOption {
```
and the type probably should be something like
```nix
with types;
  let primitive = oneOf [ str int bool float ];
  in attrsOf primitive;
```

File: modules/services/signaturepdf.nix (hunk 1 above)
Comment (DamienCassou) [reply modules/services/signaturepdf.nix:12]: indeed, sorry!

File: modules/services/signaturepdf.nix (hunk 2 above)
Comment (DamienCassou) [reply modules/services/signaturepdf.nix:28]: good idea, done.

**Comment by DamienCassou** [comment]
The build is failing because nixos-unstable doesn't contain signaturepdf yet (I just merged it this morning). I will let this PR as draft until this is solved.

**Comment by ambroisie** [comment]
@DamienCassou just curious, I see that you [just packaged it](https://github.com/NixOS/nixpkgs/pull/266336). Why didn't you create a NixOS module for it? It seems more appropriate for what is supposed to be a self-hosted service IMO.

**Comment by DamienCassou** [comment]
> Why didn't you create a NixOS module for it?

First, I (unfortunately) don't use NixOS. Second, I think that every service that can be defined in home-manager should because (1) it helps non-NixOS users and (2) it lets users control the service without requiring a root access on the machine.

**Comment by DamienCassou** [comment]
@rycee signaturepdf has been added to nixos-unstable and this PR can now be reviewed. Thank you.

**Comment by rycee** [comment]
Thanks! Merged to master now 🙂

**Pull request merged**